kotlin/edit-and-sync-features/src/main/java/com/esri/arcgisruntime/sample/editandsyncfeatures/MainActivity.kt

```diff
@@ -0,0 +1,349 @@
+        addJobDoneListener {
+          // close the progress dialog
+          generateGeodatabaseDialog.dismiss()
+          // load the geodatabase and display its feature tables on the map
+          loadGeodatabase(generateGeodatabaseJob)
+          // set edit state to ready
+          currentEditState = EditState.READY
+        }
+      }.start()
+    }
+  }
+
+  /**
+   * Loads the geodatabase from a GenerateGeodatabaseJob and displays its feature layers on the map.
+   *
+   * @param generateGeodatabaseJob the job which generated this geodatabase
+   */
+  private fun loadGeodatabase(generateGeodatabaseJob: GenerateGeodatabaseJob) {
+    // return if the job failed
+    if (generateGeodatabaseJob.status != Job.Status.SUCCEEDED) {
+      val error = generateGeodatabaseJob.error?.message ?: "Unknown error generating geodatabase"
+      Log.e(TAG, error)
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+      return
+    }
+    // if the job succeeded, load the resulting geodatabase
+    geodatabase = generateGeodatabaseJob.result
+    geodatabase?.let { geodatabase ->
+      geodatabase.loadAsync()
+      geodatabase.addDoneLoadingListener {
+        // return if the geodatabase failed to load
+        if (geodatabase.loadStatus != LoadStatus.LOADED) {
+          val error = "Error loading geodatabase: " + geodatabase.loadError?.message
+          Log.e(TAG, error)
+          Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+          return@addDoneLoadingListener
+        }
+
+        // add all of the geodatabase feature tables to the map as feature layers
+        val featureLayers =
+          geodatabase.geodatabaseFeatureTables.map { featureTable -> FeatureLayer(featureTable) }
+        mapView.map.operationalLayers.addAll(featureLayers)
+        syncButton.isEnabled = false
+      }
+    }
+  }
+
+  /**
+   * Syncs changes made on either the local or web service geodatabase with each other.
+   */
+  private fun syncGeodatabase() {
+    // create parameters for the sync task
+    val syncGeodatabaseParameters = SyncGeodatabaseParameters()
+    syncGeodatabaseParameters.syncDirection = SyncGeodatabaseParameters.SyncDirection.BIDIRECTIONAL
+    syncGeodatabaseParameters.isRollbackOnFailure = false
+    geodatabase?.let { geodatabase ->
+      // get the layer ID for each feature table in the geodatabase, then add to the sync job
+      geodatabase.geodatabaseFeatureTables.forEach { geodatabaseFeatureTable ->
+        val serviceLayerId = geodatabaseFeatureTable.serviceLayerId
+        val syncLayerOption = SyncLayerOption(serviceLayerId)
+        syncGeodatabaseParameters.layerOptions.add(syncLayerOption)
+      }
+      val syncGeodatabaseJob: SyncGeodatabaseJob = geodatabaseSyncTask
+        .syncGeodatabase(syncGeodatabaseParameters, geodatabase)
+      syncGeodatabaseJob.start()
+      val syncDialog = createProgressDialog(syncGeodatabaseJob)
+      syncDialog.show()
+      syncGeodatabaseJob.apply {
```

Comment:
Some newlines here might make it clearer all the things that are being done
```suggestion
      }
      
      val syncGeodatabaseJob: SyncGeodatabaseJob = geodatabaseSyncTask
        .syncGeodatabase(syncGeodatabaseParameters, geodatabase)
      syncGeodatabaseJob.start()
      
      val syncDialog = createProgressDialog(syncGeodatabaseJob)
      syncDialog.show()
      
      syncGeodatabaseJob.apply {
```